### PR TITLE
{devel}[GCCcore/14.2.0] Autotools v20240712, Perl v5.40.0, ...

### PR DESCRIPTION
**Pull request by Micket**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/a/Automake/Automake-1.17-GCCcore-14.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+homepage = 'https://www.gnu.org/software/automake/automake.html'
+
+description = "Automake: GNU Standards-compliant Makefile generator"
+
+toolchain = {'name': 'GCCcore', 'version': '14.2.0'}
+
+source_urls = [GNU_SOURCE]
+sources = [SOURCELOWER_TAR_GZ]
+checksums = ['07bd24ad08a64bc17250ce09ec56e921d6343903943e99ccf63bbf0705e34605']
```
Comment (branfosj) [inline easybuild/easyconfigs/a/Automake/Automake-1.17-GCCcore-14.2.0.eb:14]:
```suggestion
checksums = ['397767d4db3018dd4440825b60c64258b636eaf6bf99ac8b0897f06c89310acd']
```

File: easybuild/easyconfigs/l/libtool/libtool-2.5.4-GCCcore-14.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+description = """
+ GNU libtool is a generic library support script. Libtool hides the complexity
+ of using shared libraries behind a consistent, portable interface.
+"""
+
+toolchain = {'name': 'GCCcore', 'version': '14.2.0'}
+
+source_urls = [GNU_SOURCE]
+sources = [SOURCELOWER_TAR_GZ]
+checksums = ['04e96c2404ea70c590c546eba4202a4e12722c640016c12b9b2f1ce3d481e9a8']
```
Comment (branfosj) [inline easybuild/easyconfigs/l/libtool/libtool-2.5.4-GCCcore-14.2.0.eb:17]:
```suggestion
checksums = ['da8ebb2ce4dcf46b90098daf962cffa68f4b4f62ea60f798d0ef12929ede6adf']
```

**Comment by Micket** [comment]
@boegelbot please test @ generoso

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 12 out of 17 (9 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/fbf6eda305e99029bbad7d6635ae41c9 for a full test report.

**Comment by Micket** [comment]
I dug into this, seems like "threads" package is outdated, and is actually part of main Perl `dist` directory, with an updated working version. This was also the case in 5.38.2 already, so we probably shouldn't have built our own "threads" extension on top of it. I haven't checked earlier versions, but we should look over them as well. 

I found threads, constant, IO, Thread-Queue, Carp, Exporter, Data-Dumper so i removed those from the extensions.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 6 out of 9 (9 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/bcb88b167629c4a23128bb0d75416da7 for a full test report.

**Comment by branfosj** [comment]
https://lists.gnu.org/archive/html/bug-automake/2024-11/msg00008.html

```
I was going to suggest you could just configure for your system with:

  ./configure PERL='/usr/bin/env perl'

and it will substitute that into the scripts for you, but the configure
script in Automake 1.17 exits with a fatal error if it determines that
$PERL contains spaces.  We should probably make this non fatal since the
check is clearly too broad.  You can patch configure to not exit:

--- configure~
+++ configure
@@ -3514,7 +3514,7 @@ case $PERL in
    as_fn_error $? "perl not found" "$LINENO" 5
    ;;
  *' '* | *'    '*)
-  as_fn_error $? "The path to your Perl contains spaces or tabs.
+:  as_fn_error $? "The path to your Perl contains spaces or tabs.
 This would cause build failures later or unusable programs.
 Please use a path without spaces and try again." "$LINENO" 5
   ;;

and then it will let you set PERL='/usr/bin/env perl' and it will work.
```

**Comment by Micket** [comment]
I was just fighting with the patch for a bit. Stupid vim was expanding the tab character into spaces automatically, thus breaking the patch and it took me much longer than i care to admit to find it.

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 10 out of 11 (9 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/7391776f67304db4831852cc19b217d7 for a full test report.

**Comment by branfosj** [comment]
See https://github.com/Micket/easybuild-easyconfigs/pull/41

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 9 out of 9 (9 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/a5143d998926d09bc4e190ae3c075b7c for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen3

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 9 out of 9 (9 easyconfigs in total)
bear-pg0210u03b.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8480CL (sapphirerapids), Python 3.6.8
See https://gist.github.com/branfosj/89fdc13e6cc63322a13a1720acbab2ea for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 9 out of 9 (9 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/b719eba9408cde4792a7d103e768fe5a for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 10 out of 10 (9 easyconfigs in total)
bear-pg0201u03a.bear.cluster - Linux RHEL 8.6, x86_64, INTEL(R) XEON(R) PLATINUM 8570 (sapphirerapids), Python 3.6.8
See https://gist.github.com/branfosj/072e52e0f273511750fdaaa5d3f9df73 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 9 out of 9 (9 easyconfigs in total)
bear-pg0211u08b.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/branfosj/558dc490dbf3698a776e388aef1e4caf for a full test report.

**Pull request merged**